File: game/scripts/vscripts/items/item_boots_custom/item_boots_custom.lua
item_boots_custom = class({})
LinkLuaModifier( "modifier_item_boots_custom", "items/item_boots_custom/modifier_item_boots_custom", LUA_MODIFIER_MOTION_HORIZONTAL )

function item_boots_custom:GetIntrinsicModifierName()
    return "modifier_item_boots_custom"
end

File: src/vscripts/items/item_boots_custom.ts
import { CustomAbility } from "../abilities/framework/custom_ability";
import { CustomModifier } from "../abilities/framework/custom_modifier";
import { registerAbility, registerModifier } from "../lib/dota_ts_adapter";

@registerAbility("item_boots_custom")
class ItemBoots extends CustomAbility {
    GetIntrinsicModifierName() {
        return ModifierItemBoots.name;
    }
}

@registerModifier({ customNameForI18n: "modifier_item_boots_custom" })
class ModifierItemBoots extends CustomModifier {
    DeclareFunctions() {
        return [ModifierFunction.MOVESPEED_BONUS_PERCENTAGE];
    }

    GetModifierAttackSpeedBonus_Constant() {
        return this.Value("ms_pct");
    }
}
